node/lib/util/sequencer_state_util.js

```diff
@@ -0,0 +1,237 @@
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of git-meta nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+"use strict";
+
+const assert = require("chai").assert;
+const co     = require("co");
+const fs     = require("fs-promise");
+const path   = require("path");
+const rimraf = require("rimraf");
+
+const SequencerState = require("./sequencer_state");
+
+const CommitAndRef = SequencerState.CommitAndRef;
+
+/**
+ * This module contains methods for accessing reading and rendering
+ * `SequencerState` objects on disk.
+ */
+
+const SEQUENCER_DIR = "meta_sequencer";
+const TYPE_FILE = "TYPE";
+const ORIGINAL_HEAD_FILE = "ORIGINAL_HEAD";
+const TARGET_FILE = "TARGET";
+const COMMITS_FILE = "COMMITS";
+const CURRENT_COMMIT_FILE = "CURRENT_COMMIT";
+
+/**
+ * Return the contents of the file in the sequencer directory from the
+ * specified `gitDir` having the specified `name`, or null if the file cannot
+ * be read.
+ *
+ * @param {String} gitDir
+ * @param {String} name
+ * @return {String|null}
+ */
+exports.readFile = co.wrap(function *(gitDir, name) {
+    assert.isString(gitDir);
+    assert.isString(name);
+    const filePath = path.join(gitDir, SEQUENCER_DIR, name);
+    try {
+        return yield fs.readFile(filePath, "utf8");
+    }
+    catch (e) {
+        return null;
+    }
+});
+
+/**
+ * Read the `CommitAndRef` object from the specified `fileName` in the
+ * sequencer director in the specified `gitDir` if it exists, or null if it
+ * does not.
+ *
+ * @param {String} gitDir
+ * @param {String} fileName
+ * @return {CommitAndRef|null}
+ */
+exports.readCommitAndRef = co.wrap(function *(gitDir, fileName) {
+    assert.isString(gitDir);
+    assert.isString(fileName);
+    const content = yield exports.readFile(gitDir, fileName);
+    if (null !== content) {
+        const lines = content.split("\n");
+
+        // always an extra entry after the split
```

Comment:
Do you mean that the file consists of x\ny\n, and so split gives ['x', 'y', '']?  If so, strip the trailing newline before split to make clearer what's going on.